docker.go

```diff
@@ -182,6 +183,39 @@ func (c *DockerContainer) Name(ctx context.Context) (string, error) {
 	return inspect.Name, nil
 }
 
+// Networks gets the names of the networks the container is attached to.
+func (c *DockerContainer) Networks(ctx context.Context) ([]string, error) {
+	inspect, err := c.inspectContainer(ctx)
```

Comment:
FYI in Java, we inspect it once after the start and reuse the object, unless user asks for the current state

Reply:
Here I tried to use same pattern as in `.Name(ctx)` and `.Ports(ctx)`: inspect the container and retrieve the desired data

Reply:
I see. @gianarb perhaps you should consider doing a small refactoring, inspecting containers for each getter is an expensive operation sometimes

Reply:
btw this is what `inspectContainer` does already https://github.com/testcontainers/testcontainers-go/blob/22d6b6a6e5d8bc2a0fcad730929783ad88a3c5d2/docker.go#L155 it caches the object

Reply:
Great then! And what about https://github.com/testcontainers/testcontainers-go/issues/73#issuecomment-512865607? I could elaborate a new PR exposing the inspectContainer method as public. Wdyt, makes sense?

Reply:
@mdelapenya I would like to avoid that for now. The idea is to keep the interface as small as possible in order to able to implement the Container interface for other runtimes such as containerd.  If you expose the inspectContainer publicly it means that we need to wrap the docker response around abstraction and I would like to avoid that for now.